client.go

```diff
@@ -52,12 +56,42 @@ func GetEvaluationContextFromCtx(ctx context.Context) (ec EvaluationContext, ok
 	return ec, ok
 }
 
+func getOptionQualifiedName(opt Option) string {
+	return runtime.FuncForPC(reflect.ValueOf(opt).Pointer()).Name()
+}
+
+func isClientOption(name string) bool {
+	return strings.Contains(name, OptionWithHTTPClient) || strings.Contains(name, OptionWithRestyClient)
+}
+
 // NewClient creates instance of Client with given configuration.
 func NewClient(apiKey string, options ...Option) *Client {
 	c := &Client{
 		apiKey: apiKey,
 		config: defaultConfig(),
-		client: resty.New(),
+	}
+
+	for _, opt := range options {
+		name := getOptionQualifiedName(opt)
```

Comment:
Basically here I detect client related options based on names. They need to be ran first because some options requires the client
e.g
```
func WithCustomHeaders(headers map[string]string) Option {
	return func(c *Client) {
		c.client.SetHeaders(headers)
	}
}
```

A clean alternative would be to change the option type like 
```
type Option {
  apply: func(c *Client)
  isInit: bool
} 
```
It's quite a big refacto so let me know if it's worth going into this

Reply:
Sorry, I meant the user should not be able to set conflicting options. For example, if they are setting an HTTP client, they should not be able to set a Resty client and vice versa. Additionally, if they are passing a custom client, they should not be able to use any options that modify the client.

Reply:
Done, so now:
- Resty + Http client causes panic instead of prioritizing the resty client
- WithRequestTimeout / WithRetries / WithCustomHeaders / WithProxy will throw a panic if provided with custom clients